### PR TITLE
fix o and O undo

### PR DESCRIPTION
**Pull request by justy989**
-fix clicking on input buffer to goto it

### CODE REVIEW

File: ce_config.c [hunk 1]
```diff
@@ -1521,6 +1521,27 @@ void split_view(BufferView_t* head_view, BufferView_t* current_view, bool horizo
      }
 }
 
+void switch_to_view_at_point(ConfigState_t* config_state, Point_t point)
+{
+     BufferView_t* next_view = NULL;
+
+     if(point.x < 0) point.x = g_terminal_dimensions->x - 1;
```
Comment (ryabouza) [inline ce_config.c:1528]: This looks weird to me?
If our point (click) is left of our terminal, then move our point to the rightmost point on the terminal?
What am I missing?

File: ce_config.c [hunk 2]
```diff
@@ -2815,9 +2815,9 @@ bool key_handler(int key, BufferNode_t* head, void* user_data)
                indent_nl[indent_len + 1] = '\0';
 
                if(ce_insert_string(buffer, &begin_line, indent_nl)){
-                    // NOTE: no commit needed here, since we start the cursor on the newly added line in insert,
-                    //       mode, and the range that will be committed is inclusive
```
Comment (ryabouza) [inline ce_config.c:2819]: Lies!

Comment (justy989) [reply ce_config.c:2819]: They were !

File: ce_config.c (hunk 1 above)
Comment (robber-m) [reply ce_config.c:1528]: The points wrap. To implement ctrl+h (move window left) justin moves the cursor one to the left of the border. If we are on the left most view, this logic will wrap us around to the rightmost view

File: ce.c [hunk 3]
```diff
@@ -903,6 +903,8 @@ bool ce_get_char(const Buffer_t* buffer, const Point_t* location, char* c)
 
      *c = buffer->lines[location->y][location->x];
 
+     if(*c == 0) *c = NEWLINE;
```
Comment (robber-m) [inline ce.c:906]: ooohh

File: ce_config.c [hunk 4]
```diff
@@ -2834,9 +2834,10 @@ bool key_handler(int key, BufferNode_t* head, void* user_data)
                nl_indent[1 + indent_len] = '\0';
 
                if(ce_insert_string(buffer, &end_of_line, nl_indent)){
-                    // NOTE: no commit needed here, since we start the cursor on the newly added line in insert,
-                    //       mode, and the range that will be committed is inclusive
+                    Point_t save_cursor = *cursor;
                     *cursor = (Point_t){indent_len, cursor->y + 1};
+                    ce_commit_insert_string(&buffer_state->commit_tail, &end_of_line, &save_cursor, cursor,
```
Comment (robber-m) [inline ce_config.c:2839]: Nice

**Review by robber-m — APPROVED** [review]

**Pull request merged**